File: lib/models/message.dart
import 'user.dart';

class Message {
  final String id;
  final String challengeId;
  final User sender;
  final String content;
  final DateTime createdAt;

  Message({
    required this.id,
    required this.challengeId,
    required this.sender,
    required this.content,
    required this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['_id'] ?? '',
      challengeId: json['challenge'] ?? '',
      sender: User.fromJson(json['sender']),
      content: json['content'] ?? '',
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }
}

File: server/src/models/Message.js
const mongoose = require('mongoose');

const messageSchema = new mongoose.Schema({
  challenge: { type: mongoose.Schema.Types.ObjectId, ref: 'Challenge', required: true },
  sender: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  content: { type: String, required: true },
}, { timestamps: true });

messageSchema.index({ challenge: 1 });
messageSchema.index({ createdAt: 1 });

module.exports = mongoose.model('Message', messageSchema);
